### ISSUE TITLE
⌨  Navigation to add or edit experience is complex

### ISSUE THREAD

**Issue by RM-1978** [GCTC-NTGC/design-gc-digital-talent#19]
## ⌨ Accessibility Issue

Navigating from the Profile and applications (user dashboard) page to add an experience is too complex. First, users do not seem to understand that they must activate the "Career timeline and recruitments" link to add or edit an experience. Second, when a screen reader user lands on "Career timeline and recruitment" page, it needs to be clear where they add or edit their career experiences.

### WCAG 2.1 Criteria

[2.4.5 Multiple ways](https://w3c.github.io/wcag/understanding/multiple-ways): The intent of this Success Criterion is to make it possible for users to locate content in a manner that best meets their needs. Users may find one technique easier or more comprehensible to use than another.

## 🦋 Expected Behaviour

- Users should be able to locate the Add/Edit experiences pages in a manner that best meets their needs.
- It is clear for a user on the "Profile and applications" (user dashboard) page how to add or edit their career experiences.
- The user does not have to click 3 times to get from the "Profile and applications" (user dashboard) page to the "Add Experience" or "Edit Experience" forms.
- The way to add or edit an experience on the "Career timeline and recruitment" page are clear to a screen reader use. 

## 🕵️ Details

Add any additional details that could assist with troubleshooting/fixing the issue.
- It should be clear that "Career timeline and recruitments" card and link is where users click to manage (add or edit) their career experiences. 
- Screen reader users cannot easily activate the dialogue button for adding an experience nor the link for editing an existing experience.
- Users with low vision may find using search easier than navigating through a large menu. Users with cognitive impairments may prefer a table of contents or site map over clicking through many pages.

## 📸 Screenshots

**Profile and applications (user dashboard) page**

![Image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/1cbbee69-3cb4-4769-8202-054ffc9e587e)

**Career timeline and recruitment page**

![Image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/e18f6941-a47d-42cd-ad35-0c581948dc4c)

## 🙋‍♀️ Potential Solutions

We need to find a way to simplify the navigation flow from the "Profile and applications" to "Add/edit an experience." For example, adding two links that bypass the career timeline and recruitment page.  In general, provide multiple ways to reach any page on a site allows users to chose whatever way of finding pages is easiest for them.

## ✅ Acceptance Criteria (to be modified based on solutions)

A set of assumptions which, when tested, verify that the issue was addressed.

- [ ] Users should be able to locate the Add/Edit experiences pages in multiple ways.
- [ ] It is clear for a user on the "Profile and applications" (user dashboard) page how to add or edit their career experiences.
- [ ] The user does not have to click 3 times to get from the "Profile and applications" (user dashboard) page to the "Add Experience" or "Edit Experience" forms.
- [ ] The way to add or edit an experience on the "Career timeline and recruitment" page are clear to a screen reader use.

**Comment by substrae** [comment]
I understand the goal, yes.

Number of clicks isn't necessarily a good measurement. The task should be performed in context and where the task is the primary behaviour. That said, @gobyrne @Jerryescandon, was keeping the "type" dialog in this flow on purpose? My designs removed it from the flow entirely to reduce the number of interactions, as the creation form asks you to choose a type. This reduces the creation to a two click task.

In terms of adding a button to the dashboard - I'm not convinced this task warrants this level of attention. In the grand scheme of things, creating an experience isn't one of the behaviours our users will be performing frequently (compared to managing applications, updating personal information, and managing their skills). This page has already been flagged for new design work to accommodate the next group of work stream features, so the goal then will be to drastically reduce the amount of information in favor of providing quick access to the most prominent behaviours.

**Comment by gobyrne** [comment]
Thanks for the discussion. Here are my thoughts/actions.

- I'll make an issue to add a sitemap or similar to address 2.4.5 across the site.
- While we can agree that the flow isn't as simple as we would like for adding an experience, it's a product of us having a lot to show and do on this page. Let's try and do better next time we redesign this page (already planned).
- On this specific page, the flow affects everyone and 2.4.5 isn't about making all designs perfect. So this page isn't failing on that criteria for this reason.
- I'm closing this issue in favor of above actions.
